soh/Makefile

```diff
@@ -24,18 +24,18 @@ WARN := \
 	-funsigned-char \
 	-fno-stack-protector -fno-common -fno-zero-initialized-in-bss -fno-strict-aliasing -fno-inline-functions -fno-inline-small-functions -ffreestanding -fwrapv \
 
-CXXFLAGS := $(WARN) -std=c++20 -D_GNU_SOURCE -fpermissive -no-pie -nostdlib
-CFLAGS := $(WARN) -std=c99 -D_GNU_SOURCE -no-pie -nostdlib
+CXXFLAGS := $(WARN) -std=c++20 -D_GNU_SOURCE -fpermissive -no-pie -nostdlib $(shell pkg-config --cflags sdl2 glew) $(shell sdl2-config --cflags)
+CFLAGS := $(WARN) -std=c99 -D_GNU_SOURCE -no-pie -nostdlib $(shell pkg-config --cflags sdl2 glew) $(shell sdl2-config --cflags)
 LDFLAGS :=
 
-ifeq ($(UNAME), Linux) #LINUX
-	CXXFLAGS += -mhard-float -msse2 -mfpmath=sse
-	CFLAGS += -mhard-float -msse2 -mfpmath=sse
-endif
+ifeq ($(UNAME), Linux)
+	ifeq ($(UNAMEM), x86_64)
+		CXXFLAGS += -msse2 -mfpmath=sse -mhard-float
```

Comment:
these kind of scare me a bit tbh, i know they were there before, but they make debugging a bit harder, and in this case they're on for debug builds.

for "release" on people's own system, maybe `-march=native -mcpu=native` would be best? and for release from the CI/us, we can probably come up with -march, -mtune, -mcpu, etc. for different architectures which are more permissive than native, and should work across popular cpus

Reply:
I have to be honest and say that I don't really understand these flags.. so I get the local one we can use: `-march=native -mcpu=native` and that should choose the appropriate SIMD extensions right?

but what about for the general case, what are the appropriate flags for ARM and x86_64?